src/com/intellij/idea/plugin/hybris/tools/remote/http/AbstractHybrisHacHttpClient.java

```diff
@@ -111,22 +111,50 @@ public String login(@NotNull Project project, @NotNull HybrisRemoteConnectionSet
     }
 
     @NotNull
-    public final HttpResponse post(@NotNull Project project, @NotNull String actionUrl, @NotNull List<BasicNameValuePair> params, boolean canReLoginIfNeeded) {
+    public final HttpResponse post(
+        @NotNull Project project,
+        @NotNull String actionUrl,
+        @NotNull List<BasicNameValuePair> params,
+        boolean canReLoginIfNeeded,
+        final long timeout
+    ) {
+        return post(project, actionUrl, params, canReLoginIfNeeded, 6000L);
+    }
+
+    @NotNull
+    public final HttpResponse post(
+        @NotNull Project project,
+        @NotNull String actionUrl,
+        @NotNull List<BasicNameValuePair> params,
+        boolean canReLoginIfNeeded
+    ) {
+        return post(project, actionUrl, params, canReLoginIfNeeded, 6000L, null);
+    }
+
+    @NotNull
+    public final HttpResponse post(
+        @NotNull Project project,
+        @NotNull String actionUrl,
+        @NotNull List<BasicNameValuePair> params,
+        boolean canReLoginIfNeeded,
+        final long timeout,
+        HybrisRemoteConnectionSettings settings
+    ) {
         if (sessionId == null) {
             String errorMessage = login(project);
             if (StringUtils.isNotBlank(errorMessage)) {
                 return createErrorResponse(errorMessage);
             }
         }
-        String csrfToken = getCsrfToken(getHostHacURL(project), sessionId);
+        String csrfToken = getCsrfToken(getHostHacURL(project), sessionId, project, settings);
         if (csrfToken == null) {
             this.sessionId = null;
             if (canReLoginIfNeeded) {
                 return post(project, actionUrl, params, false);
             }
             return createErrorResponse("Unable to obtain csrfToken for sessionId="+sessionId);
         }
-        HttpClient client = createAllowAllClient(6000L);
```

Comment:
If we have 6 sec as default - please make them as constant and min start value in the spinner.